src/Core/src/Platform/Android/ViewExtensions.cs

```diff
@@ -199,6 +203,39 @@ public static void UpdateBackground(this AView platformView, Paint? background)
 			}
 		}
 
+		// TODO: NET7 make this public for net7.0
+		internal static void UpdateBackground(this AView platformView, Paint? background, bool keepPreviousDrawable)
+		{
+			var paint = background;
+
+			if (!paint.IsNullOrEmpty())
+			{
+				// Remove previous background gradient if any
+				if (platformView.Background is MauiDrawable mauiDrawable)
+				{
+					platformView.Background = null;
+					mauiDrawable.Dispose();
+				}
+
+				if (keepPreviousDrawable)
```

Comment:
Is there any reason to not just make this how the default behavior works? The current behavior seems like a bug (especially coming from XF) so I think we can just make this how it works and we don't need an overload. Ideally we'd add an API so users could just set the underline color

Reply:
No good reasons, inherited behavior from Xamarin.Forms. @Redth What do you think about this?, should we replace the default behavior?

Reply:
XF does _not_ change the underline color when the background color changes. However, I believe that's because the Entry in XF is based on `VisualElementRenderer`, so the background being changed is actually the wrapper.